.pre-commit-config.yaml

```diff
@@ -31,3 +31,13 @@ repos:
       args: [--allow-redefinition]
       exclude: ^examples/
       additional_dependencies: [types-tqdm, types-Pillow]
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: v0.8.6
```

Comment:
Would we need to remove `flake8`, `isort` and `black`? The documentation says it has drop-in parity.